core/frontend/src/LocalhostIpcApp.ts

```diff
@@ -12,20 +12,27 @@ import { WebViewerApp, WebViewerAppOpts } from "./WebViewerApp";
 
 class LocalTransport extends IpcWebSocketTransport {
   private _client: WebSocket;
+  private _pending?: IpcWebSocketMessage[] = [];
 
   public constructor(port: number) {
     super();
 
     this._client = new WebSocket(`ws://localhost:${port}/`);
 
+    this._client.addEventListener("open", () => {
+      const pending = this._pending!;
+      this._pending = undefined;
+      pending.forEach((m) => this.send(m));
```

Comment:
@swbsi should you drop the listener here?

Reply:
> @swbsi should you drop the listener here?

The socket instance can't be opened again, so I have never bothered to.